test/takeWhile.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+
+test('takeWhile', function() {
+
+  eq(typeof S.takeWhile, 'function');
+  eq(S.takeWhile.length, 2);
+  eq(S.takeWhile.toString(), 'takeWhile :: (Foldable f, Monoid f) => (a -> Boolean) -> f a -> f a');
+
+  function lt(b) {
+    return function(a) {
+      return a < b;
+    };
+  }
+
+  eq(S.takeWhile(lt(1), [1, 2, 3, 4, 5]), []);
+  eq(S.takeWhile(lt(2), [1, 2, 3, 4, 5]), [1]);
+  eq(S.takeWhile(lt(3), [1, 2, 3, 4, 5]), [1, 2]);
+  eq(S.takeWhile(lt(4), [1, 2, 3, 4, 5]), [1, 2, 3]);
+  eq(S.takeWhile(lt(5), [1, 2, 3, 4, 5]), [1, 2, 3, 4]);
+  eq(S.takeWhile(lt(6), [1, 2, 3, 4, 5]), [1, 2, 3, 4, 5]);
```

Comment:
Let's change these to demonstrate and assert that `takeWhile` is different from `filter`.